framework/src/controller/channels/child_process_channel.ts

```diff
@@ -46,62 +42,86 @@ export const rejectWhenTimeout = async (timeInMillis: number): Promise<void> =>
 
 type NodeCallback = (error: Error | null, result?: unknown) => void;
 
-const SOCKET_TIMEOUT_TIME = 2000;
+interface ChildProcessOptions extends BaseChannelOptions {
+	socketsPath: SocketPaths;
+}
 
 export class ChildProcessChannel extends BaseChannel {
-	public localBus: EventEmitter2;
-	public subSocket?: SubEmitterSocket;
-	public busRpcSocket?: ReqSocket;
-	public busRpcClient?: Client;
-	public busRpcClientCallPromisified: unknown;
-	public pubSocket?: PubEmitterSocket;
-	public rpcSocketPath?: SocketPaths | string;
-	public rpcSocket?: RepSocket;
-	public rpcServer?: RPCServer;
+	private readonly _emitter: EventEmitter2;
+	private readonly _ipcClient: IPCClient;
 
 	public constructor(
 		moduleAlias: string,
 		events: ReadonlyArray<string>,
 		actions: ActionsDefinition,
-		options: BaseChannelOptions = {},
+		options: ChildProcessOptions,
 	) {
 		super(moduleAlias, events, actions, options);
-		this.localBus = new EventEmitter2();
+
+		this._ipcClient = new IPCClient({
+			socketsDir: options.socketsPath.root,
+			name: moduleAlias,
+			rpcServerSocketPath: `unix://${options.socketsPath.root}/bus_rpc_socket.sock`,
+		});
+
+		this._emitter = new EventEmitter2({
```

Comment:
do we still need to use EventEmitter2?

Reply:
Yes, we need a local emitter. We can use native `EventEmitter` but it does not support wildcard. If one wants to listen `channel.subscribe('ap:*')` it can't be done with native `EventEmitter`. If we don't want that support anymore then we can remove `EventEmitter2`

Reply:
Change to push/pull will not affect the wildcard feature, then no need to remove them